### PR TITLE
VACMS-17323: Updates module with fixes for button

### PR DESCRIPTION
**Pull request by omahane**
## Description

Relates to #17325

## Testing done
Manually

### Office Hours (set)
Screenshot 2024-03-08 at 4 42 28 PM

### Office Hours (not set)
Screenshot 2024-03-08 at 4 45 11 PM

### Buttons restored
Screenshot 2024-03-08 at 4 46 51 PM

### Set up test user
- [x] Log in as an admin and set QA Content Publisher to the following:
  - [x] Roles:
    - [x] Content publisher
    - [x] Content creator - VAMC
    - [x] Content creator - VBA
    - [x] Content creation - Vet Center
  - [x] Sections:
    - [x] VA Boston health care 
    - [x] VA Illiana health care
    

#### Vet Center - Community Access Point 
- [ ] Go to [Fargo Vet Center - Bagley, MN](https://pr17451-3coflbqwjllvbyoghxxvvqqwwrsnq0mg.ci.cms.va.gov/fargo-vet-center/community-access-point/fargo-vet-center-bagley-mn)
- [x] Click **Edit**
- [x] Click **Provide CAP hours online**
- [x] Add hours:
  - [ ] Monday - Friday: 8 am - 4:30 pm
  - [ ] Leave Saturday and Sunday blank
- [x] Save the node in the same moderation state as it is currently
- [x] Click the node title to view the node
- [x] Confirm that the hours are displayed correctly
  - [ ] The filled in hours are accurate
  - [ ] The hours not filled in show as "Closed"
  
#### Vet Center - Mobile Vet Center
- [ ] Go to [Abilene Mobile Vet Center](https://pr17451-3coflbqwjllvbyoghxxvvqqwwrsnq0mg.ci.cms.va.gov/abilene-vet-center/abilene-mobile-vet-center)
- [x] Confirm that the hours are showing
- [x] Click Edit
- [x] Confirm that the hours are showing
- [x] Confirm that the hours are not editable

#### Vet Center
- [ ] Go to [Abilene Vet Center](https://pr17451-3coflbqwjllvbyoghxxvvqqwwrsnq0mg.ci.cms.va.gov/abilene-vet-center)
- [x] Confirm that the hours are showing
- [x] Click Edit
- [x] Confirm that the hours are showing
- [x] Confirm that the hours are not editable

#### Vet Center - Outstation
- [ ] Go to [Grand Forks Vet Center Outstation](https://pr17451-3coflbqwjllvbyoghxxvvqqwwrsnq0mg.ci.cms.va.gov/fargo-vet-center/grand-forks-vet-center-outstation)
- [x] Confirm that the hours are showing
- [x] Click Edit
- [x] Confirm that the hours are showing
- [x] Confirm that the hours are not editable

#### VAMC Facility
- [ ] Go to [Quincy VA Clinic](https://pr17451-3coflbqwjllvbyoghxxvvqqwwrsnq0mg.ci.cms.va.gov/boston-health-care/locations/quincy-va-clinic)
- [x] Confirm that the hours are showing
- [x] Click Edit
- [x] Confirm that the hours are showing
- [x] Confirm that the hours are not editable

#### VBA Facility
- [ ] Go to [Cheyenne VA Regional Benefit Office](https://pr17451-3coflbqwjllvbyoghxxvvqqwwrsnq0mg.ci.cms.va.gov/cheyenne-va-regional-benefit-office/locations/cheyenne-va-regional-benefit-office)
- [x] Confirm that the hours are showing
- [x] Click Edit
- [x] Confirm that the hours are showing
- [x] Confirm that the hours are not editable

#### VAMC System Billing and Insurance
- [ ] Go to [Boston VA health care: Billing and Insurance](https://pr17451-3coflbqwjllvbyoghxxvvqqwwrsnq0mg.ci.cms.va.gov/boston-health-care/billing-and-insurance)
- [x] Confirm that the hours are showing (_Note: they are in 24-time_, not something to correct here.): M-F Open, S&S Closed.
- [x] Click Edit
- [x] Confirm that the Mon - Fri hours are filled in with the same values
- [x] Confirm that the Sat - Sun hours are not filled in

#### VAMC Facility Health Service
- [x] Go to [Amputation care - Jamaica Plain VA Medical Center](https://pr17451-3coflbqwjllvbyoghxxvvqqwwrsnq0mg.ci.cms.va.gov/node/30310/edit)
- [x] Click **Hours** to open accordion
- [x] Click **Provide specific hours for this service**
- [x] Add hours:
  - [x] Monday - Friday: 8 am - 4:30 pm
  - [x] Leave Saturday and Sunday blank
- [x] Save the node in the same moderation state as it is currently
- [x] Click the node title to view the node
- [x] Confirm that the hours are displayed correctly
  - [x] The filled in hours are accurate
  - [x] The hours not filled in show as "Closed"
- [x] Edit the facility
- [x] Click **Hours** to open accordion
- [x] Click **Use the facility's hours**
- [x] Save the node
- [x] Confirm that the hours have changed

### Check NCA Facility as an admin
- [x] Log in as an admin
- [x] Go to a NCA Facility
- [x] Confirm that hours are showing
- [x] Click Edit
- [x] Confirm that hours are filled in

# Check any front-end effects
- [ ] Do the queries return anything different from graphql?
- [ ] Does the front-end look different?

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.
- [ ] If there are field changes, front end output has been thoroughly checked.

### Select Team for PR review

- [ ] `CMS Team`
- [ ] `Public websites`
- [x] `Facilities`
- [ ] `User support`
- [ ] `Accelerated Publishing`

### CODE REVIEW

**Comment by davidmpickett** [comment]
Main feedback is that you missed a couple places where the Office Hours widget is used. I made [a PR to add those](https://github.com/department-of-veterans-affairs/va.gov-cms/pull/17470).

I double-checked the [Content Model Field report](https://pr17451-3coflbqwjllvbyoghxxvvqqwwrsnq0mg.ci.cms.va.gov/admin/reports/content-model/fields?field_name=field_office_hours&field_type=&entity_type=&bundle=&label=) to ensure we covered all entities that use field_office_hours. 

Also just just learned about the ["used in views" field report](https://pr17451-3coflbqwjllvbyoghxxvvqqwwrsnq0mg.ci.cms.va.gov/admin/reports/fields/views-fields) and confirmed that the non_clinical_services view is the only one we need to consider.

**Comment by davidmpickett** [comment]
@omahane Just found an unintended consequence of showing empties. On the Service Location Paragraph type, if you've selected use the Facility Hours it will show that correctly, but then also show a block of closed hours because the custom hours field is a separate field that is always there, but just hidden out of sight.

Screenshot 2024-03-11 at 9 49 48 AM

Screenshot 2024-03-11 at 9 44 46 AM

Not sure if there's a way to have the Node View only should the Custom hours if the corresponding radio button is selected?

File: config/sync/core.entity_view_display.node.vba_facility.default.yml [hunk 1]
```diff
@@ -436,6 +436,7 @@ content:
       time_format: g
       compress: false
       grouped: false
+      show_empty: false
```
Comment (davidmpickett) [inline config/sync/core.entity_view_display.node.vba_facility.default.yml:439]: I think this should be true

Comment (omahane) [inline config/sync/core.entity_view_display.node.vba_facility.default.yml:439]:
```suggestion
      show_empty: true
```

**Review by omahane — COMMENTED** [review]
Change that.

**Review by Becapa — APPROVED** [review]
This looks good to me!

**Review by davidmpickett — COMMENTED** [review]
Perfect. Thanks!

**Comment by omahane** [comment]
@eselkin This has already been merged, but I'd still like to confirm that we won't have any regressions for office hours on the front end. Would you be able to help me test this?

**Pull request merged**